crates/sui-tool/src/replay.rs

```diff
@@ -436,24 +438,54 @@ impl LocalExec {
         Ok(o)
     }
 
-    pub fn download_latest_object(&self, object_id: &ObjectID) -> Result<Object, LocalExecError> {
-        // TODO: replace use of `block_on`
-        block_on(self.download_latest_object_impl(object_id))
+    // TODO: remove this after `futures::executor::block_on` is removed.
+    #[allow(clippy::disallowed_methods)]
+    pub fn download_latest_object(
+        &self,
+        object_id: &ObjectID,
+    ) -> Result<Option<Object>, LocalExecError> {
+        block_on({
+            info!("Downloading latest object {object_id}");
+            self.download_latest_object_impl(object_id)
+        })
     }
 
     pub async fn download_latest_object_impl(
         &self,
         object_id: &ObjectID,
-    ) -> Result<Object, LocalExecError> {
+    ) -> Result<Option<Object>, LocalExecError> {
         let options = SuiObjectDataOptions::bcs_lossless();
-        let object = self
+
+        match self
             .client
             .read_api()
             .get_object_with_options(*object_id, options)
             .await
-            .map_err(|q| LocalExecError::SuiRpcError { err: q.to_string() })?;
-
-        obj_from_sui_obj_response(&object)
+        {
+            Ok(obj) => match obj_from_sui_obj_response(&obj) {
```

Comment:
I think you could use `map_ok` or `and_then` so you don't have these two duplicated `Err(_)` cases